src/webgpu/api/operation/render_pass/storeOp.spec.ts

```diff
@@ -1,23 +1,352 @@
+
+    t.device.defaultQueue.submit([encoder.finish()]);
+
+    // Check that the correct store operation occurred.
+    let expectedColorValue: PerTexelComponent<number> = {};
+    if (t.params.colorStoreOperation === 'clear') {
+      // If colorStoreOp was clear, the texture should now contain {0.0, 0.0, 0.0, 0.0}.
+      expectedColorValue = { R: 0.0, G: 0.0, B: 0.0, A: 0.0 };
+    } else if (t.params.colorStoreOperation === 'store') {
+      // If colorStoreOP was store, the texture should still contain {1.0, 0.0, 0.0, 1.0}.
+      expectedColorValue = { R: 1.0, G: 0.0, B: 0.0, A: 1.0 };
+    }
+    t.expectSingleColor(colorAttachment, kColorFormat, {
+      size: [kHeight, kWidth, 1],
+      exp: expectedColorValue,
+    });
+
+    // Check that the correct store operation occurred.
+    let expectedDepthValue: PerTexelComponent<number> = {};
+    if (t.params.depthStencilStoreOperation === 'clear') {
+      // If depthStencilStoreOperation was clear, the texture's depth component should be 0.0, and
+      // the stencil component should be 0.0.
+      expectedDepthValue = { Depth: 0.0 };
+    } else if (t.params.depthStencilStoreOperation === 'store') {
+      // If depthStencilStoreOperation was store, the texture's depth component should be 1.0, and
+      // the stencil component should be 1.0.
+      expectedDepthValue = { Depth: 1.0 };
+    }
+    t.expectSingleColor(depthStencilAttachment, kDepthStencilFormat, {
+      size: [kHeight, kWidth, 1],
+      exp: expectedDepthValue,
+    });
+  });
+
+// Tests that render pass color attachment store operations work correctly for all renderable color
+// formats, mip levels and array layers.
+g.test('render_pass_store_op,color_attachment_only')
+  .params(
+    params()
+      .combine(poptions('colorFormat', kTextureFormats))
+      // Filter out any depth/stencil or non-renderable formats
+      .unless(
+        ({ colorFormat }) =>
+          !kTextureFormatInfo[colorFormat].color || !kTextureFormatInfo[colorFormat].renderable
+      )
+      .combine(poptions('storeOperation', kStoreOps))
+      .combine(poptions('mipLevel', kMipLevel))
+      .combine(poptions('arrayLayer', kArrayLayers))
+  )
+  .fn(t => {
+    const kColorFormat: GPUTextureFormat = 'rgba8unorm';
+    const colorAttachment = t.device.createTexture({
+      format: kColorFormat,
+      size: { width: kWidth, height: kHeight, depth: t.params.arrayLayer + 1 },
+      mipLevelCount: t.params.mipLevel + 1,
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.OUTPUT_ATTACHMENT,
+    });
+
+    const colorViewDesc: GPUTextureViewDescriptor = {
+      format: kColorFormat,
+      dimension: '2d',
+      aspect: 'all',
+      baseArrayLayer: t.params.arrayLayer,
+      baseMipLevel: t.params.mipLevel,
+      arrayLayerCount: 1,
+      mipLevelCount: 1,
+    };
+
+    const colorAttachmentView = colorAttachment.createView(colorViewDesc);
+
+    // Color load operation will clear to {1.0, 0.0, 0.0, 1.0}.
```

Comment:
Can we have all non-zeros? It would help make sure all components are correctly written.